### PR TITLE
Attributes, inter-communicators, and spawning

### PR DESCRIPTION
**Pull request by jedbrown**
This isn't super full-featured, but I think the interface is convenient enough. Process spawning with Open MPI is a bit more expensive than launching using `mpiexec`.

### CODE REVIEW

**Comment by jtronge** [comment]
This looks great. I tried the three examples locally.

Is this assuming that inter communicators are created outside of rsmpi right now? I didn't see any `MPI_Intercomm_*` functions.

**Comment by jedbrown** [comment]
`MPI_Comm_spawn` creates intercommunicators. I didn't bind `MPI_Intercomm_create` because I'm interested in spawning, but it could be done later.

File: Cargo.toml [hunk 1]
```diff
@@ -27,14 +27,15 @@ derive = ["mpi-derive"]
 
 [dependencies]
 conv = "0.3"
-libffi = { version = "3.0.0", optional = true }
+libffi = { version = "3.0.1", optional = true }
```
Comment (hppritcha) [inline Cargo.toml:30]: were these version changes needed for this PR or just to pull in newer versions?

Comment (jedbrown) [reply Cargo.toml:30]: Not necessary; could be dropped.

**Pull request merged**